harvard-lil/capstone-static
# Add initial CSS Variables and Map placeholder component

## What this does 
Adds initial baseline styles, and an index page, for the static overhaul of the Caselaw Access Project website. Specifically: 

- Adds a web component, `caselaw-map`, to test out how well global css variables will apply to encapsulated web components. The web component also includes some placeholder text that will be removed when we pull in actual CAP data.
- Adds global CSS variables: 
  - Colors 
  - Fonts
  - Spacing
  - Typography
- Adds css reset in `wc-base.js`
- Adds all font families
- Adds baseline `index.html`, with some CAP metadata (no open graph images just yet) 

## What it should look like
![Screenshot 2023-10-17 at 10 54 22 AM](https://github.com/harvard-lil/capstone-static/assets/4039311/9815c240-f4a1-4f22-b6c4-7a9e5406a28c)

## Implementation notes 
- There are a lot of files added, but fear not: most of them are fonts!
- The `caselaw-map` component uses Lit primarily for easier readability. [Lit](https://lit.dev/) is a minimal, well-documented library maintained by Google, and incorporated into design systems including Adobe's Spectrum Web Components, IBM's Carbon Web Components, and JSTOR'S Pharos Design System. 
- By default, all web components that are scaffolded using Lit's base element use the Shadow DOM. This can be disabled, but it would mean that we couldn't natively scope CSS to a web component, which I think is one of the big benefits of them. I really appreciate the readability of being able to see JavaScript, HTML, and CSS in the same view/same file for a component.
- The CSS Reset is an exported module in `wc-base.js` so that it can be shared between all web components, [following Lit's code example. ](https://lit.dev/docs/components/styles/#sharing-styles). 
- CSS Variables don't need to be encapsulated in the same way because CSS Variables bleed through the Shadow DOM by design

## How to test
- Check out the remote fork and try to run the website locally. There shouldn't be any dependencies to install. 
  - I've been using the [Live Server](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer) extension to spin up the website, but because all the files are static, you should also be able to load the `index.html` file itself in whatever way is most convenient for you. 
- Check: 
  - Does the gradient display? 
  - Do the fonts load?
    - You may see a difference in font appearance between Firefox and Arc/Chrome. As far as I can tell, anti-aliasing fonts inside of a web component is not considered valid in Firefox, which is a bit confusing, because it is considered valid in Firefox for non-web components.
